private/utils.rkt

```diff
@@ -0,0 +1,12 @@
+#lang racket
+;;; aux funcs.
+(provide (all-defined-out))
+
+;; NOTE: assumes `m` is a (Listof (List KeyType ValueType)).
+;; this func will not set! for you; you have to do it manually.
+(define (assocmap-set m k v)
+  (cond
+    ((null? m) m)
```

Comment:
use racket coding style: `(cond (c e) ...)` -> `(cond [c e] ...)`

Reply:
Though really, you can just write 

```
(dict-set alist k (list v))
```

instead. No need to define a new function.